gm4_smelteries/data/gm4_smelteries/functions/smelt.mcfunction

```diff
@@ -0,0 +1,10 @@
+# @s - @e[type=armor_stand,tag=gm4_bf_valid_structure,tag=gm4_bf_could_smelt] if found cooked furnace
+# called by tick
+
+data modify entity @s HandItems[0] set from entity @s ArmorItems[3]
+loot spawn ~ ~1 ~ fish gm4_smelteries:smelt ~ ~ ~ mainhand
+replaceitem entity @s weapon air
+
```

Comment:
BUG : `gm4_bf_could_smelt` never gets removed from the entity. So as long as a blast furnace has ever has ever been used properly will no longer require the structure around it to function. 

How to fix this bug without requiring the blast furnace to go through expensive checks over and over though I haven't figured out yet.

Reply:
I've figured out what I think is a actually pretty efficient way of removing the tag by restricting the conditions when it receives the tag and removing after a smelt attempt.